### PR TITLE
Accounts

### PR DESCRIPTION
**Pull request by sb3958**
Account Page
The Accounts Page
Deactivate Pop Up Message
The Deactivate Pop Up
Delete Pop Up Message
The Delete Pop Up

### CODE REVIEW

File: frontend/src/FullPageRoutes/AccountsPage.tsx [hunk 1]
```diff
@@ -0,0 +1,263 @@
...
+    },
+    topBar: {
+        margin: 1,
+        padding: 50,
+        backgroundColor: "rgb(238,235,228)",
+    },
+    pageContent: {
+        margin: 8,
+        padding: 110,
+    },
```
Comment (mneedleman) [inline frontend/src/FullPageRoutes/AccountsPage.tsx]: No unitless css please

File: frontend/src/FullPageRoutes/AccountsPage.tsx [hunk 2]
```diff
@@ -0,0 +1,263 @@
...
+    const [btnText, setBtnText] = useState('Delete or Deactivate Account')
+    const [open, setOpen] = useState(false)
+    const [accounts, setAccounts] = useState( )
+    
+       
+
+    const handleChange = () => {
+
+    }
+
```
Comment (mneedleman) [inline frontend/src/FullPageRoutes/AccountsPage.tsx]: Don't need empty functions

File: frontend/src/FullPageRoutes/AccountsPage.tsx [hunk 3]
```diff
@@ -0,0 +1,263 @@
...
+                    InputProps={{
+                        endAdornment:(
+                            <InputAdornment position="end">
+                                <IconButton onClick={handleEdit}>
+                                    <EditIcon/>
+                                </IconButton>
+                            </InputAdornment>
+                        ),
+                        
+                    }}
```
Comment (mneedleman) [inline frontend/src/FullPageRoutes/AccountsPage.tsx]: There are clearly some spacing issues here. Please run this through the linter

**Review by AyalaCnaan — CHANGES_REQUESTED** [review]
This looks pretty encouraging, but without a screenshot it's hard to say much. Please provide a screenshot.

File: frontend/src/FullPageRoutes/AccountsPage.tsx [hunk 4]
```diff
@@ -0,0 +1,263 @@
...
+                    <Button type="submit" variant="contained" className={classes.button} onClick={handleAccountChanges}>{btnText}</Button>
+                    <Dialog open={open} onClose={handleClose}>
+                        <DialogContent>
+                            Are you sure you to {btnText}?
+                        </DialogContent>
+                        <DialogActions>
+                            <Button onClick={handleClose}>Yes</Button>
+                            <Button onClick={handleClose}>No</Button>
+                        </DialogActions>
+                    </Dialog>
```
Comment (mneedleman) [inline frontend/src/FullPageRoutes/AccountsPage.tsx]: Let's use the `AbstractModal` that we have in the `components` folder.

**Review by mneedleman — CHANGES_REQUESTED** [review]

**Comment by sb3958** [comment]
I provided screenshots of how it look.

File: frontend/src/FullPageRoutes/AccountsPage.tsx [hunk 5]
```diff
@@ -0,0 +1,263 @@
+import * as React from 'react';
+import { Grid, makeStyles, Button, Typography, TextField, InputAdornment, IconButton, Divider, FormControl, FormControlLabel, FormLabel, RadioGroup, Radio, FormHelperText, Dialog, DialogContent, DialogActions} from '@material-ui/core';
+import EditIcon from '@material-ui/icons/Edit';
+import { useState, useEffect } from 'react'
+
+export type userAccounts={
```
Comment (elp49) [inline frontend/src/FullPageRoutes/AccountsPage.tsx]: capitalize the first letter of types

File: frontend/src/FullPageRoutes/AccountsPage.tsx [hunk 6]
```diff
@@ -0,0 +1,263 @@
...
+        }
+    }
+
+    
+    const handleClose = () =>{
+        setOpen(false);
+    }
+
+    useEffect(()=>{
+        React.request({ path: "/api/profile", method: "get" })
```
Comment (elp49) [inline frontend/src/FullPageRoutes/AccountsPage.tsx]: Since we've decided to make Account its own page we will want to add an account endpoint but that might be for another PR

File: api/views.py [hunk 7]
```diff
@@ -5,8 +5,13 @@
 from django.db.models import Q
 from rest_framework import generics, permissions, request, response, status, views
 
-from .models import Video, Profile
+<<<<<<< HEAD
+from .models import Profile, Video
+=======
+from .models import Video
+>>>>>>> 2dad5bcba2cc2e628357aaebb61550d1d2be88ce
```
Comment (mneedleman) [inline api/views.py]: This looks like it's left over from a merge conflict. Please fix

File: frontend/src/App.tsx [hunk 8]
```diff
@@ -6,27 +6,34 @@ import TokenPage from "./FullPageRoutes/TokenPage"
...
+import AccountsPage from "./FullPageRoutes/AccountsPage"
 
 const useStyles = makeStyles({
   header: {
     backgroundColor: "#cd0b2d",
     height: "3rem",
     padding: "0 1rem",
+    position: "fixed",
+    left: 0,
+    top: 0,
```
Comment (mneedleman) [inline frontend/src/App.tsx]: No unitless css please

File: frontend/src/App.tsx [hunk 9]
```diff
@@ -83,12 +90,12 @@ export default function App(): JSX.Element {
...
-              <VideoListPage />
-            </Route>
             <Route path="/Profile">
               <ProfilePage />
             </Route>
+            <Route path="/Account">
+              <AccountsPage />
+            </Route>
             <Route path="/">
               <MainPage />
```
Comment (mneedleman) [inline frontend/src/App.tsx]: This looks out of date. Did you pull?

File: api/views.py [hunk 10]
```diff
@@ -5,7 +5,11 @@
 from django.db.models import Q
 from rest_framework import generics, permissions, request, response, status, views
 
+<<<<<<< HEAD
 from .models import Profile, Video
+=======
+from .models import Video
+>>>>>>> 2dad5bcba2cc2e628357aaebb61550d1d2be88ce
```
Comment (elp49) [inline api/views.py]: these changes were not merged

**Review by mneedleman — CHANGES_REQUESTED** [review]
You need to pull

**Review by elp49 — DISMISSED** [review]

**Review by AyalaCnaan — DISMISSED** [review]
It looks good to me.

**Review by jcarrete5 — COMMENTED** [review]
Do we want to keep the default testing values e.g. Jane Doe and the dummy email? Seems strange to leave have default values for these fields. What purpose do they serve?

**Review by mneedleman — APPROVED** [review]
Looks good to me 🚢 :shipit:
